packages/eslint-plugin/docs/rules/no-require-imports.md

```diff
@@ -28,6 +28,28 @@ import { lib2 } from 'lib2';
 import * as lib3 from 'lib3';
 ```
 
+## Options
+
+### `allowPackageJson`
```

Comment:
I've also seen libraries want to import from `turbo.json`, `project.json`, etc. Maybe this should be the standard format from https://github.com/typescript-eslint/typescript-eslint/discussions/6017, with docs showing how to use it for `package.json` files? WDYT?

Reply:
I'm okay with that. What's your proposed API? My thought is something similar to `no-restricted-imports`, with an array of `paths`.

Reply:
I was thinking the exact format from  https://github.com/typescript-eslint/typescript-eslint/pull/4436 -> https://typescript-eslint.io/rules/prefer-readonly-parameter-types.

```json
{
  "@typescript-eslint/no-var-requires": [
    "error",
    {
      ignore: [
        {
          name: "package.json",
          source: "file",
        },
      ],
    },
  ],
}
```

Reply:
This looks couunterintuitive as a user, TBH. This API is for _types_, but what is `"package.json"` as a type? What does `source: "file"` even mean?

Reply:
What happens if they want to import from a node module, e.g. `import emojis from "emojilib/dist/emoji-en-US.json"`? Should we just have a `string[]` as the option to capture all those cases? Feels off to me that we have the big thought-out format and then don't use it... but it is overhead, yes. 🤔

Reply:
ping @Josh-Cena ?

Reply:
Sorry, I did not forget this 😅 Will try to garner some energy and clear my notifications this afternoon.

Reply:
@JoshuaKGoldberg I thought about it. Here are my thoughts:

1. The only thing this PR changes is allowing `require`. It doesn't change anything about `import`—they can still be used as normal.
2. Sure, we have a thought-out format for specifying _types_, but here we are talking about import _paths_, and this design doesn't come without its precedent either—we are being consistent with `no-restricted-imports`.
3. About having `string[]` as an option. You mean `{ ignorePaths: string[] }`? My initial design is `{ ignorePaths: TheOptionsOfNoRestrictedImports }`, i.e. it supports multiple ignore paths, each one with a path, a glob, a custom message, etc. But I now think that's hugely unnecessary for an _ignore_ option. So sure, I can do that.
4. If you want to differentiate between node modules and local JSON files, you would have to support regex in the ignore path (I think) to be able to tell between `./foo/data.json` and `foo/data.json`. That means the option has to be more granular than a `string[]`; it would at minimum need to be `{ ignorePaths: { paths: string[] /* globs */, patterns: string[] /* regexes */ }`. Does that make sense? Am I overthinking it?

I'm very curious what your current idea for the option's shape is.

Reply:
The intent of https://github.com/typescript-eslint/typescript-eslint/discussions/6017 was: _"RFC: Common format to specify a type or value as a rule option"_. I was really hoping it'd be usable for situations exactly like this one. But if we're seeing a practical shortcoming of the format -it being too heavy-handed- then I like this as a motivation to improve it. Or maybe it's just irrelevant here...

Maybe just `allow: string[]` or `ignore: string[]`, where each of the string elements is a regex? It'd be a little verbose to have to say something like `"package\.json$"]`, but not the worst thing in the world IMO.

Thinking on the sticking point of relativity: from what I can think of either:
* the rule gets told a root ESLint path to resolve from _(but what about monorepos with config in the root vs per-package?)_
* we tell users to either increase regex specificity or allow sibling level `package.json` imports (`/[.\\\/]*package.json/`)

I think having two things to think about, like `paths` and `patterns` from 4., would be a bit much. Is there anything that can only be captured in that system, not the single option?

Reply:
The complicated thing is that sometimes you have tsconfig paths which make things look non-relative. We can't build in any specific assumptions around the path here if we want people to allow local but disallow node_module packages done this way.

So a regex is probably the most flexible option for this so codebases can easily devise some regex that matches their usecase. I'd suspect that most codebases would just want something like you suggested joshg to allow a relative package.json